src/bp-groups/classes/class-bp-groups-component.php

```diff
@@ -555,10 +555,10 @@ public function setup_nav( $main_nav = array(), $sub_nav = array() ) {
 		$bp = buddypress();
 
 		// Determine user to use.
-		if ( bp_displayed_user_domain() ) {
-			$user_domain = bp_displayed_user_domain();
-		} elseif ( bp_loggedin_user_domain() ) {
+		if ( bp_loggedin_user_domain() ) {
 			$user_domain = bp_loggedin_user_domain();
+		} elseif ( bp_displayed_user_domain() ) {
+			$user_domain = bp_displayed_user_domain();
```

Comment:
@ngolwalkar Still, we can't make this change as it will create issues. I would suggest you make the change on that menu URL if someone added in the menu then add by logged in user in the menu link.